frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts

```diff
@@ -54,18 +60,33 @@ export class FeatureFlagOverviewDetailsSectionCardComponent {
     if (slideToggleEvent.checked) {
       this.openEnableConfirmModel();
     } else {
-      // this.openDisableConfirmModel();
-      console.log('disable!');
+      this.openDisableConfirmModel();
     }
 
-    // Note: we don't want the toggle to change state immediately because we have to pop a confirmation modal first, so we need override the default and flip it back
+    // Note: we don't want the toggle to visibly change state immediately because we have to pop a confirmation modal first, so we need override the default and flip it back. I unfortunately couldn't find a better way to do this.
     slideToggleEvent.checked = !slideToggleEvent.checked;
   }
 
   openEnableConfirmModel(): void {
-    this.dialogService.openEnableFeatureFlagConfirmModel();
-    //remove this?
-    this.featureFlag$.subscribe((featureFlag) => console.log({ featureFlag }));
+    const dialogRef = this.dialogService.openEnableFeatureFlagConfirmModel();
+
+    dialogRef.afterClosed().subscribe(() => {
+      this.changeDetectorRef.detectChanges();
+    });
+  }
+
+  openDisableConfirmModel(): void {
+    const dialogRef = this.dialogService.openDisableFeatureFlagConfirmModel();
+
+    dialogRef.afterClosed().subscribe(() => {
+      this.changeDetectorRef.detectChanges();
+    });
+  }
+
+  listenForClosedModal(dialogRef: MatDialogRef<UpdateFlagStatusConfirmationModalComponent>): void {
+    dialogRef.afterClosed().subscribe(() => {
+      this.changeDetectorRef.detectChanges();
```

Comment:
I didn't understand the need for this `this.changeDetectorRef.detectChanges();` and why listen to it separately and in the model sub. can you share why?

Reply:
hmI I think that looks like something that I left in as I was trying for debugging getting the chip-status to update after enable, but it didn't work. I'll remove them.